crates/wasmtime/src/runtime/code_memory.rs

```diff
@@ -322,23 +322,31 @@ impl CodeMemory {
 
                     #[cfg(has_virtual_memory)]
                     {
-                        let text = self.text();
-
-                        use wasmtime_jit_icache_coherence as icache_coherence;
-
-                        // Clear the newly allocated code from cache if the processor requires it
-                        //
-                        // Do this before marking the memory as R+X, technically we should be able to do it after
-                        // but there are some CPU's that have had errata about doing this with read only memory.
-                        icache_coherence::clear_cache(text.as_ptr().cast(), text.len())
+                        #[cfg(any(not(windows), feature = "std"))]
```

Comment:
For this it's a pretty critical operation here to manage the icache so we'll want to avoid this `#[cfg]` which makes it difficult to see when it's used and when it isn't. I think the best solution here might be to sidestep the problem entirely and update [this line](https://github.com/bytecodealliance/wasmtime/blob/5ac3f799820812b70b88411d9e9ac1833a98069c/crates/wasmtime/build.rs#L12) to take the `std` feature into account. For example that could be `(unix || windows) && cfg!(feature = "std")` or something like that. A comment there would be good to explain that if `std` is disabled then it's not supported in Wasmtime right now to interact with the OS which means that it's no longer supported.

Reply:
I tried that change but it doesn't solve the problem: Wasmtime still pulls `std` through `jit-icache-coherence` in `no_std` Windows.
Another possible solution is to use `libc` in `jit-icache-coherence` for `no_std` Windows at this [line](https://github.com/bytecodealliance/wasmtime/blob/5ac3f799820812b70b88411d9e9ac1833a98069c/crates/jit-icache-coherence/src/lib.rs#L82). For that, we could add a new `std` feature flag to `jit-icache-coherence` and enable it if `std` in `wasmtime` is enabled.

Reply:
I just pushed a [new commit](https://github.com/bytecodealliance/wasmtime/pull/11152/commits/ceae3ffa2e00abd0aa1086c55c22f4cce10e7019) that implements the change suggested above but happy to implement a different solution if this is problematic.

Reply:
Your commit looks reasonable, thanks for doing that! I'll apologize again though about how this is a bit messy. I'd prefer to avoid having the crate be sort of half-std/half-not, so I think I've come up with an idea to sidestep this entirely which is to include the icache crate only if `std` is enabled in Wasmtime itself. That solves the concerns I have about things getting a bit weird sometimes and the only remaining issue is various CI bits and how well this works out. I've pushed up a commit to this PR which implements what I was thinking of and I'll watch CI to see if my strategy won't work due to something I haven't thought of